java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -267,4 +238,35 @@ public void info_with_empty_InfoOptions_returns_success() {
         assertEquals(testResponse, response);
         assertEquals(testPayload, payload);
     }
+
+    private static Stream<Arguments> getCommandsWithoutArgs() {
+        return Map.<RequestType, Function<RedisClient, CompletableFuture<String>>>of(
+                        ConfigRewrite, RedisClient::configRewrite,
+                        ConfigResetStat, RedisClient::configResetStat,
+                        Info, RedisClient::info,
+                        Ping, RedisClient::ping)
+                .entrySet()
+                .stream()
+                .map(e -> Arguments.of(e.getKey(), e.getValue()));
+    }
+
+    @SneakyThrows
+    @ParameterizedTest(name = "{0} returns success")
+    @MethodSource("getCommandsWithoutArgs")
+    public void no_arg_command_returns_success(
+            RequestType requestType, Function<RedisClient, CompletableFuture<String>> command) {
+        // setup
+        CompletableFuture<String> testResponse = mock(CompletableFuture.class);
+        when(testResponse.get()).thenReturn(OK);
+        when(commandManager.<String>submitNewCommand(eq(requestType), eq(new String[0]), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<String> response = command.apply(service);
+        String payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(OK, payload);
```

Comment:
Info and ping return strings rather than OK, but this is a minor detail and maybe not worth worrying about if we're able to remove some code duplication here like you've shown

Reply:
I reverted this and added new tests